src/mjolnir/graphenhancer.cc

```diff
@@ -49,19 +45,93 @@ using namespace valhalla::mjolnir;
 
 namespace {
 
-// Geometry types for admin queries
-typedef boost::geometry::model::d2::point_xy<double> point_type;
-typedef boost::geometry::model::polygon<point_type> polygon_type;
-typedef boost::geometry::model::multi_polygon<polygon_type> multi_polygon_type;
-
 // Number of tries when determining not thru edges
 constexpr uint32_t kMaxNoThruTries = 256;
 
 // Radius (km) to use for density
 constexpr float kDensityRadius = 2.0f;
-constexpr float kDensityRadius2 = kDensityRadius * kDensityRadius;
 constexpr float kDensityLatDeg = (kDensityRadius * kMetersPerKm) / kMetersPerDegreeLat;
 
+struct DensityCellId {
+  // Density grid operates with cell_id derived from coordinate by shifting its bits by (32 -
+  // kGridLevel), meaning that bigger level produces smaller cells.
```

Comment:
why `32 - level`? why not just 2 raised to the power of the level, `1 << level`, like normal tiling systems?

Reply:
:pray: i very much appreciate it!